### ISSUE TITLE
Cannot instantiate `std::hash_map` with a `void` value type

### ISSUE THREAD

**Issue by ashn-dot-dev** [ashn-dot-dev/sunder#118]
```sunder
import "std";

func main() void {
    # This compiles fine.
    var map = std::hash_map[[s32, s32]]::init(std::null_allocator::ALLOCATOR);

    # This does not compile!
    var map = std::hash_map[[s32, void]]::init(std::null_allocator::ALLOCATOR);
}
```

```sh
$ sunder-compile test.sunder
[std/core.sunder:343] error: struct `std::optional[[std::key_value_pair[[s32, void]]]]` contains a member variable of incomplete struct type `std::key_value_pair[[s32, void]]`
    var _value: T;
    ^
```

**Comment by ashn-dot-dev** [comment]
This appears to happen with any zero-sized value struct:

```sunder
import "std";

struct foo {}

func main() void {
    var x = (:std::key_value_pair[[s32, foo]]){.key = 123, .value = std::void::init()};
}
```

```sh
$ sunder-compile test.sunder
[test.sunder:6] error: struct `std::key_value_pair[[s32, foo]]` is incomplete
    var x = (:std::key_value_pair[[s32, foo]]){.key = 123, .value = std::void::init()};
            ^
```

**Comment by ashn-dot-dev** [comment]
Fixed in bac98b9afc56da436da541d675d85e61c7d0399a.
